### ISSUE TITLE
Cannot delegate to an existing static method

### ISSUE THREAD

**Issue by dsyer** [raphw/byte-buddy#449]
I might be doing it wrong, but I don't immediately see what. If I want to add a static method that calls another static method, it seems to be doomed to fail, at least if I use the existing code in `MethodCall`. Here's what I have currently in my transformer:

```java
.intercept(
	MethodCall
			.invoke(named("run")
					.and(isStatic()
					.and(takesArguments(Object[].class, String[].class))
					.and(isVarArgs())))
			.withAllArguments());
```

and I can see that the filter doesn't pick any existing methods, despite the fact that there is one. The reason, I believe, is that in `MethodCall` bytebuddy delegates to `MethodGraph.Compiler.Default.compile(TypeDefinition, TypeDescription)` and in there there is a filter that only picks virtual methods (i.e. it discards all static methods). So how do I select the implementation of an existing static method, since I don't see a way to change that filter?

Also see SO question: https://stackoverflow.com/questions/49956509/add-a-static-method-to-class-and-delegate-to-an-existing-method.

**Comment by raphw** [comment]
I answered your question on StackOverflow. Please reopen this ticket if you have any further questions.

**Comment by dsyer** [comment]
Thanks. I managed to get it to work. I can't say it was very obvious how to do it, even after reading all the javadocs. It's quite a big API. Maybe some more samples, or snippets in the README would help? I also had a lot of trouble working out how to select an existing constructor (for essentially the same reason). I feel dumb now that I couldn't do it quicker, but I did spend more than half a day thrashing around on this, not realizing why it wasn't working.
